generator/src/main/java/org/corfudb/generator/LongevityApp.java

```diff
@@ -0,0 +1,99 @@
+package org.corfudb.generator;
+
+import org.corfudb.generator.operations.CheckpointOperation;
+import org.corfudb.generator.operations.Operation;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by rmichoud on 7/25/17.
+ */
+
+/**
+ * This Longevity app will stress test corfu using the Load
+ * generator during a given duration.
+ *
+ * Current implementation spawns a task producer and a task
+ * consumer. Both of them are a single threaded. In combination
+ * with the blocking queue, they limit the amount of task created
+ * during execution. Apps thread pool will take care of executing
+ * the operations.
+ *
+ */
+public class LongevityApp {
+
+    private long durationMs;
+    private String configurationString;
+
+    public LongevityApp(long durationMs, String configurationString) {
+        this.durationMs = durationMs;
+        this.configurationString = configurationString;
+    }
+
+    public void runLongevityTest() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)